### PR TITLE
[#139693829] Photos list on photos page

### PR DESCRIPTION
**Pull request by clearlyTHUYDOAN**
I worked on hooking up photos and displaying them in a list.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    // Use IntelliSense to learn about possible Node.js debug attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (melaniebrgr) [inline .vscode/launch.json]: Can this file be removed from the PR?

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { get } from '../api/request';
+
+export const ACTION_TYPES = {
```
Comment (melaniebrgr) [inline src/actions/index.js:3]: Can we split the action types and actions into different files?

File: src/app/app.js [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
-function App() {
-  return (
-    <div>
-      Hello world!
-    </div>
-  );
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
+
+import { getPhotos } from '../actions/index.js';
```
Comment (melaniebrgr) [inline src/app/app.js]: Don't need to include the '.js' at the end of your local module imports. The build/bundler tool is smart enough to know its javascript if you leave it off.

File: src/app/app.js [hunk 4]
```diff
@@ -1,12 +1,35 @@
...
+import Image from '../components/image.js';
+
+const mapStateToProps = state => ({
```
Comment (melaniebrgr) [inline src/app/app.js]: Totally optional, but you can use destructuring here, e.g.
`const mapStateToProps = { photos } => ({ photos });`

Comment (melaniebrgr) [inline src/app/app.js]: I forgot you can return objects from arrow functions only if you wrap in parens -- nice.

File: src/app/app.js [hunk 5]
```diff
@@ -1,12 +1,35 @@
...
+  }
 }
 
+App.propTypes = {
+  getPhotos: React.PropTypes.func,
+  photos: React.PropTypes.array,
+};
 
-export default App;
+export default connect(mapStateToProps, mapDispatchToProps)(App);
```
Comment (melaniebrgr) [inline src/app/app.js]: Another personal preference but I like keep the mapStateToProps mapDispatchToProps function and object near the connect function their used in.

File: src/app/app.js [hunk 6]
```diff
@@ -1,12 +1,35 @@
...
+            <h3 className="center-align">Photo Map App</h3>
+            <Image photos={this.props.photos}/>
+          </div>
+    );
+  }
 }
 
+App.propTypes = {
+  getPhotos: React.PropTypes.func,
+  photos: React.PropTypes.array,
```
Comment (melaniebrgr) [inline src/app/app.js]: Consider making the photos prop required? Maybe getPhotos too, not sure.

File: src/app/app.js [hunk 7]
```diff
@@ -1,12 +1,35 @@
...
+  getPhotos,
+};
+
+class App extends Component {
+  componentDidMount() {
+    this.props.getPhotos();
+  }
+  render() {
+    return (
+          <div className="container">
```
Comment (melaniebrgr) [inline src/app/app.js]: As we talked about, let's pull out all styling for now.

File: src/app/app.js [hunk 8]
```diff
@@ -1,12 +1,35 @@
...
+
+const mapStateToProps = state => ({
+  photos: state.photos.list,
+});
+
+const mapDispatchToProps = {
+  getPhotos,
+};
+
+class App extends Component {
```
Comment (melaniebrgr) [inline src/app/app.js]: Instead of App, let's make this it's own, more specific component... PhotoGallery? or just Gallery?

File: src/app/app.js [hunk 9]
```diff
@@ -1,12 +1,35 @@
-import React from 'react';
-
-function App() {
-  return (
-    <div>
-      Hello world!
-    </div>
-  );
+import React, {Component} from 'react';
```
Comment (melaniebrgr) [inline src/app/app.js]: Spaces around Component, just to be similar to the others `{ Component }`

File: src/components/image.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+const Image = (props) => (
```
Comment (melaniebrgr) [inline src/components/image.js]: You can destructure props here too, e.g. `const Image = { photos } => (...`

File: src/components/image.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    <div className="row">
+    {props.photos.map((image, index) => {
+      return (
+        <div key={index} className="col s4">
```
Comment (melaniebrgr) [inline src/components/image.js]: Can be broken down into smaller components. Maybe leave out the card-content for now and create another component when you get to that story.

File: src/components/image.js (hunk 10 above)
Comment (melaniebrgr) [reply src/components/image.js]: Might need parens around the curly braces here^

File: src/index.html [hunk 12]
```diff
@@ -5,6 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="description" content="">
     <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.98.0/css/materialize.min.css">
```
Comment (melaniebrgr) [inline src/index.html]: Maybe just comment out for now.

File: src/reducers/photos.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+const INITIAL_STATE = {
+  list: [],
+};
+
+// Reducer
+export const photos = (state = INITIAL_STATE, {type, payload}) => {
+  switch (type) {
+  case ACTION_TYPES.getPhotos:
+    return {...state, ...{list: payload.photos}};
```
Comment (melaniebrgr) [inline src/reducers/photos.js:11]: Object destructuring! Awesome!

**Review by melaniebrgr — CHANGES_REQUESTED** [review]
Such a great start Thuy! We are definitely hitting the ground running thanks to this.

File: .vscode/launch.json [hunk 14]
```diff
@@ -1,7 +1,4 @@
 {
```
Comment (melaniebrgr) [inline .vscode/launch.json:1]: Can .gitignore?

**Review by melaniebrgr — CHANGES_REQUESTED** [review]

**Review by melaniebrgr — APPROVED** [review]

**Review by melaniebrgr — APPROVED** [review]

File: src/reducers/photos.js (hunk 13 above)
Comment (melaniebrgr) [reply src/reducers/photos.js:11]: I was wrong about this^. It's actually an example of [object spread operator](http://redux.js.org/docs/recipes/UsingObjectSpreadOperator.html) use.

**Comment by pkanal** [comment]
Had to merge this as admin because something wasnt right with Circle CI. Will look into that separately.

**Pull request merged**